### PR TITLE
Add phase referencing

### PR DESCRIPTION
**Pull request by darafferty**

### CODE REVIEW

File: rapthor/lib/screen.py [hunk 1]
```diff
@@ -334,17 +334,25 @@ def fit(self):
         vals = [[ra*np.pi/180.0, dec*np.pi/180.0] for ra, dec in zip(ra_deg, dec_deg)]
         sourceTable = list(zip(*(soltab_ph.dir, vals)))
 
-        # Now call LoSoTo's stationscreen operation to do the fitting
+        # Now call LoSoTo's stationscreen operation to do the fitting. For the phase
+        # screens, we reference the the phases to the station with the least amount
```
Comment (gmloose) [inline rapthor/lib/screen.py:338]: Twice "the"

Comment (darafferty) [reply rapthor/lib/screen.py:338]: Good catch!

File: rapthor/lib/screen.py [hunk 2]
```diff
@@ -536,8 +545,19 @@ def fit(self):
...
         # Input data are [time, freq, ant, dir, pol] for slow amplitudes
-        # and [time, freq, ant, dir] for fast phases (scalarphase).
+        # and [time, freq, ant, dir] for fast phases (scalarphase). We reference
+        # the phases to the station with the least amount of flagged solutions,
+        # drawn from the first 10 stations (to ensure it is fairly central)
+        weights = soltab_ph.getValues(retAxesVals=False, weight=True)
+        weights = np.sum(weights, axis=tuple([i for i, axis_name in
+                                              enumerate(soltab_ph.getAxesNames())
+                                              if axis_name != 'ant']), dtype=np.float)
+        ref_ind = np.where(weights[0:10] == np.max(weights[0:10]))[0][0]
```
Comment (gmloose) [inline rapthor/lib/screen.py:555]: This block is (AFAICS) identical to lines 341-345. You might want to refactor this into a function.

File: rapthor/lib/screen.py [hunk 3]
```diff
@@ -334,17 +334,25 @@ def fit(self):
         vals = [[ra*np.pi/180.0, dec*np.pi/180.0] for ra, dec in zip(ra_deg, dec_deg)]
         sourceTable = list(zip(*(soltab_ph.dir, vals)))
 
-        # Now call LoSoTo's stationscreen operation to do the fitting
+        # Now call LoSoTo's stationscreen operation to do the fitting. For the phase
+        # screens, we reference the the phases to the station with the least amount
+        # of flagged solutions, drawn from the first 10 stations (to ensure it is
+        # fairly central)
```
Comment (gmloose) [inline rapthor/lib/screen.py:340]: Is it guaranteed that the stations with the lowest index are also the most central?

**Review by gmloose — COMMENTED** [review]
I cannot judge for the processing correctness, but I added a few comments.

File: rapthor/lib/screen.py (hunk 3 above)
Comment (darafferty) [reply rapthor/lib/screen.py:340]: With LOFAR at least I think it is (at least in my experience).

File: rapthor/lib/screen.py (hunk 2 above)
Comment (darafferty) [reply rapthor/lib/screen.py:555]: Good idea, will do.

**Pull request merged**